install

```diff
@@ -41,86 +49,127 @@ while [ : ]
+
+    waiting_pod_array=("k8s-app=kube-dns;kube-system" 
+                       "k8s-app=metrics-server;kube-system"
+                       "app=traefik;kube-system"  
+                       "app=minio;kubeflow"
+                       "app=mysql;kubeflow"
+                       "app=cache-server;kubeflow"
+                       "app=ml-pipeline-persistenceagent;kubeflow"
+                       "component=metadata-grpc-server;kubeflow"
+                       "app=ml-pipeline-ui;kubeflow")
+
+    for i in "${waiting_pod_array[@]}"; do 
+      echo "$i"; 
+      IFS=';' read -ra VALUES <<< "$i"
+        wait "${VALUES[0]}" "${VALUES[1]}"
+    done
+
+
+
+    info "Kubeflow pipelines ready!!"
+
+    info "Defining the ingress"
+    sleep_cursor
+
+    kubectl apply -f - << EOF
+      apiVersion: networking.k8s.io/v1beta1
+      kind: IngressClass
+      metadata: 
+        name: traefik-lb
+      spec: 
+        controller: traefik.io/ingress-controller
+EOF
 
-if [ $1 = "--gpu" ]
-  then   
-    info "Install Pipelines with GPU support, docker as runtime"
-    curl -sfL https://get.k3s.io | sh -s - --docker --kubelet-arg="feature-gates=DevicePlugins=true" --write-kubeconfig-mode 644
-    #### GPU Support
-    kubectl create -f https://raw.githubusercontent.com/NVIDIA/k8s-device-plugin/v0.7.0/nvidia-device-plugin.yml
-  else  
-    info "Install Pipelines without GPU support, containerd as runtime"
-    curl -sfL https://get.k3s.io | sh -s - --write-kubeconfig-mode 644
-fi
-
-export KUBECONFIG=/etc/rancher/k3s/k3s.yaml
-
-info "Installing pipelines crd"
-export PIPELINE_VERSION=1.0.1
-kubectl apply -k "github.com/kubeflow/pipelines/manifests/kustomize/cluster-scoped-resources?ref=$PIPELINE_VERSION"
-kubectl wait --for condition=established --timeout=60s crd/applications.app.k8s.io
-sleep_cursor &
-info "Installing pipelines manifests"
-kubectl apply -k "github.com/kubeflow/pipelines/manifests/kustomize/env/platform-agnostic-pns?ref=$PIPELINE_VERSION"
-
-waiting_pod_array=("k8s-app=kube-dns;kube-system" 
-                   "k8s-app=metrics-server;kube-system"
-                   "app=traefik;kube-system"  
-                   "app=minio;kubeflow"
-                   "app=mysql;kubeflow"
-                   "app=cache-server;kubeflow"
-                   "app=ml-pipeline-persistenceagent;kubeflow"
-                   "component=metadata-grpc-server;kubeflow"
-                   "app=ml-pipeline-ui;kubeflow")
-
-for i in "${waiting_pod_array[@]}"; do 
-  echo "$i"; 
-  IFS=';' read -ra VALUES <<< "$i"
-    wait "${VALUES[0]}" "${VALUES[1]}"
-done
+    kubectl apply -f - << EOF
+      apiVersion: "networking.k8s.io/v1beta1"
+      kind: "Ingress"
+      metadata:
+        name: "pipeline-ingress"
+        namespace: kubeflow
+        annotations:
+          nginx.ingress.kubernetes.io/rewrite-target: /$2
+          
+      spec:
+        ingressClassName: "traefik-lb"
```

Comment:
so it's using traefik...but you still need istio for kubeflow?

Reply:
You are probably right but it might be worth having something simpler than istio in kubeflow...I guess we can see if there's interest in the kubeflow community

Reply:
Hi @rossella 
Thank you for your feedback.

Kubeflow pipeline does not need istio. 
traefik is enabled by default in k3s, this is why we are using it.
For full Kubefow installation, we will remove traefik for istio